Реализация gophkeeper с tui

k0st1a opened the pull request:
На текущий момент не все сделано, но т.к. времени осталось мало, то, думаю, текущего кода будет достаточно для принятия проекта.

Что сделано:
* серверная сторона по работе к предмета/юзера gophkeeper-а
* локальное хранилище предметов клиента
* синхронизация предметов клиента и сервера
* tui

Что хотелось сделать, но не сделано:
* шифрование серверных предметов 
* паджинация при синхронизации предметов (сейчас вычитывается все за раз)
* паджинацию предметов для tui (сейчас вычитывается все за раз)
* юнит и интеграционные тесты (не успеваю по сроку)
* github ci для запуска тестов
* github ci для сборки релиза
* потоковую передачу файлов
* прикрутить к серверу S3 хранилище

Моменты:
* от swagger решил отказаться, много времени на него потратил
* tui работает - проверял ручками)
* тесты (юнит и интеграционные) напишу в дальнейшем, без них не комильфо)

Как запустить:
* запускаем db командой `make db-up`
* командой `make` собираем клиента и сервера
* командой `make server-run-with-args` запускаем сервер
* командой `make client-run-with-args` запускаем tui клиента. Нужно зарегистрироваться, залогиниться и посоздавать/удалять/обновить предметы.
* P.S. командой `make client-run-with-args-2` можно запустить 2-ого клиента

k0st1a committed:
Добавлено техническое задание для проекта

k0st1a committed:
Поддержка базовой конфигурации для сервера

k0st1a committed:
Добавление пакета grpcserver

k0st1a committed:
Добавление пакета logwrap

k0st1a committed:
Добавлено описание proto файла для auth и сгенерированный на его основе код

k0st1a committed:
Добавлен скелет запуска сервера

k0st1a committed:
Добавлен golangci-lint

k0st1a committed:
Добавлены go.mod и go.sum

k0st1a committed:
Правки по результатам запуска golangci-lint

k0st1a committed:
Добавлена миграция создания таблицы users + API таботы с users

k0st1a committed:
В логгер добавлен traceid

k0st1a committed:
Добавлен запуск/остановка контейнера с PostgreSQL

k0st1a committed:
login заменен на email

k0st1a committed:
Добавлен пакет для сервисного слоя регистрации и логина пользователя

k0st1a committed:
Добавлен пакет auth для работы с токеном и паролем

k0st1a committed:
Перенос/переименование proto файла + правка для reverse proxy

k0st1a committed:
Добавлены правкила для установки и генерации go, grpc, gateway, openapi файлов

k0st1a committed:
Добавлена генерация swagger-ui

github.com/johanbrandhorst/grpc-gateway-boilerplate

k0st1a committed:
Добавлено правило запуска сервера

k0st1a committed:
Поправлено расположение сгенерированных файлов из proto

k0st1a committed:
Поправлена конфигурация сервера

k0st1a committed:
Поддержка сервисного слоя работы с пользователем + запуск reverse proxy и swagger

k0st1a committed:
Актуализированы go.mod и go.sum

k0st1a committed:
Правки по результатам запуска golangci-lint

k0st1a committed:
В Makefile добавлено правило cover - оценка покрытия тестами

k0st1a committed:
Поправлен Makefile для генерации items и users

k0st1a committed:
Поправлена миграция users

k0st1a committed:
Добавлена миграцию items

k0st1a committed:
Переименование протофайла + его правки

k0st1a committed:
Добавлено protobuf описание items

k0st1a committed:
Удален сгенерированный auth за ненадобностью

k0st1a committed:
Реализация grpc кода для серверной стороны

k0st1a committed:
Актуализация работы с БД на серверной стороне

k0st1a committed:
Добавлен пакет userid

k0st1a committed:
Актуализация application-а server и gateway

k0st1a committed:
Актуализация third_party

k0st1a committed:
Обноавлен README

k0st1a committed:
Код удален за ненадобностью

k0st1a committed:
Добавлены сгенерированные файлы

k0st1a committed:
Начало работ по клиенту

k0st1a committed:
Добавлено правило компиляции/запуска клиента

k0st1a committed:
Добавлено grpc для клиентской стороны

k0st1a committed:
Добавлено вычитывание конфиги клиента

k0st1a committed:
Поправлено вычитывание списка итемов с сервера

k0st1a committed:
Доработки grpc-шной части клиента

k0st1a committed:
Начало работ по cli клиента

k0st1a committed:
Правки после запуска statictest

k0st1a committed:
Мелкая правка

k0st1a committed:
Начало работ над TUI

k0st1a committed:
Добавлено логирование в файл для клиента

k0st1a committed:
Логирование даже когда логера нет в контексте

k0st1a committed:
В Makefile добавлена команда установки easyjson

k0st1a committed:
Поправлено пробрасывание порта для докера с PostgreSQL

k0st1a committed:
Изменен внешний порт докера с PosgreSQL

k0st1a committed:
Поправлен запуск и контроль PostgreSQL в докере

k0st1a committed:
Поправлен Makefile:
* поправленs пути генерации proto файлов;
* удалена генерация swagger-ui
* добавлен запуск клиента 2

k0st1a committed:
Обновлены go.mod и go.sum

k0st1a committed:
Удалено third_party

k0st1a committed:
Удалено scripts

k0st1a committed:
Перенесены и обновлены proto файлы

k0st1a committed:
Порты для серверной стороны перенесены в папку server

k0st1a committed:
Обновлена работа db хранилища

k0st1a committed:
Обновлены сгенерированные proto файлы

k0st1a committed:
Удален за ненадобностью пакет openapitool

k0st1a committed:
Обновлен grpc сервер

k0st1a committed:
Добавлены модели для стороны клиента

k0st1a committed:
Добавлены порты для клиентской стороны

k0st1a committed:
Удален gateway за ненадобностью

k0st1a committed:
Добавлен inmemory storage для клиентской стороны

k0st1a committed:
Обновлен grpc клиент

k0st1a committed:
Добавлен пакет sync для синхронизации между локальным хранилищем клиента и удаленным хранилищем на сервере

k0st1a committed:
Добавлен пакет tick для запуска периодической сихронизации предметов клиента и сервера

k0st1a committed:
Добавлен пакет job для контроля (start/stop) запущенной gorutine-ы

В частности нужно для контроля тикера по которому запускается
синхронизация предметов между клиентом и сервером

k0st1a committed:
cli удален за ненадобностью

k0st1a committed:
Добавлен клиент для tui для взаимодействия с локальным хранилищем клиента

k0st1a committed:
Добавлен tui

k0st1a committed:
Поправлен запуск севера

k0st1a committed:
Поправлен запуск клиента

k0st1a committed:
Поправлен error выписы

k0st1a committed:
Переименовано полу Id в ID, т.к. не проходили тесты

k0st1a committed:
Добавлено исключение easyjson в cover

Чтобы код easyjson не учитывался в % покрытия тестов

k0st1a committed:
Правки после запуска golangci-lint

k0st1a committed:
В клиент и сервер добавлен вывод buildVersion, buildDate, buildCommit

k0st1a committed:
Правки после запуса golangci-lint

k0st1a committed:
Правки после запуса golangci-lint

k0st1a committed:
Правки после запуса golangci-lint

k0st1a committed:
Правки после запуса golangci-lint

k0st1a committed:
Правки после запуса golangci-lint

k0st1a committed:
Правки после запуса golangci-lint: использование ошибок вместо строк

k0st1a committed:
Правки после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
Правки после запуса golangci-lint: использование констант вместо magic number

k0st1a committed:
Правки после запуса golangci-lint: использование констант вместо magic numbers

k0st1a committed:
Правки после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
Правки после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
Правки после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
Добавлены комментарии к полям структуры

k0st1a committed:
Правки после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
В конфиг клиента добавлены параметры RequestTimeout и SyncInterval

k0st1a committed:
Мелкая правка лога

k0st1a committed:
Правки после запуса golangci-lint: использование констант вместо magic number

k0st1a committed:
Правка после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
Добавлены комментарии к полям структуры

k0st1a committed:
Правка после запуса golangci-lint: теперь структура занимает меньше места

k0st1a committed:
Правки после запуса golangci-lint

k0st1a committed:
Добавлен тесты для tui/storage/client

k0st1a committed:
Добавлен тест с использование mockgen

k0st1a committed:
Добавлены тесты

k0st1a committed:
Add изменение на Set + покрытие тестом пакета userid

avhitrov commented on internal/pkg/client/model/item.go:
```diff
@@ -0,0 +1,85 @@
+	Card     *Card     `json:"card"`
+	Password *Password `json:"password"`
+	Note     *Note     `json:"note"`
+	File     *File     `json:"file"`
```
Позволю себе зацитировать ТЗ:
```
Для любых данных должна быть возможность хранения произвольной текстовой метаинформации (принадлежность данных к веб-сайту, личности или банку, списки одноразовых кодов активации и прочее).
```

k0st1a replied:
Поправил в рамках коммита [9be99a11961984bee009f3d29136b29e73dde5ac](https://github.com/k0st1a/gophkeeper/pull/1/commits/9be99a11961984bee009f3d29136b29e73dde5ac)

k0st1a replied:
Касательно защиты, можно попробовать)

avhitrov reviewed (changes requested):
Практически отлично!
- Есть одно забытое текстовое поле, которое имеет смысл протащить на уровне модели Item. В остальном придраться особо не к чему.
- Есть предложение подумать о защите проекта.

k0st1a committed:
Добавлены поля Meta для использования метаинформации о предмете

Правка по результатам Code Review https://github.com/k0st1a/gophkeeper/pull/1#pullrequestreview-2400129235

avhitrov reviewed (approved):
Все ок. Поздравляю с завершением дипломной работы!
